pkg/lib/cockpit-components-file-autocomplete.jsx

```diff
@@ -152,50 +138,42 @@ export class FileAutoComplete extends React.Component {
         });
     }
 
-    onToggle(_, isOpen) {
-        this.setState({ isOpen });
-    }
-
     clearSelection() {
-        this.typeaheadInputValue = "";
         this.updateFiles("/");
-        this.setState({
-            value: null,
-            isOpen: false
-        });
+        this.setState({ value: null });
         this.props.onChange('', null);
     }
 
     render() {
         const placeholder = this.props.placeholder || _("Path to file");
 
         const selectOptions = this.state.displayFiles
-                .map(option => <SelectOption key={option.path}
-                                             className={option.type}
-                                             value={option.path} />);
+                .map(option => ({ value: option.path, content: option.path, className: option.type }));
+
         return (
-            <Select
-                variant="typeahead"
-                id={this.props.id}
-                isInputValuePersisted
-                onTypeaheadInputChanged={this.debouncedChange}
-                placeholderText={placeholder}
-                noResultsFoundText={this.state.error || _("No such file or directory")}
-                selections={this.state.value}
-                onSelect={(_, value) => {
-                    this.setState({ value, isOpen: false });
-                    this.debouncedChange(value);
-                    this.props.onChange(value || '', null);
-                }}
-                onToggle={this.onToggle}
-                onClear={this.clearSelection}
-                isOpen={this.state.isOpen}
-                isCreatable={this.props.isOptionCreatable}
-                createText={_("Create")}
-                onCreateOption={this.onCreateOption}
-                menuAppendTo="parent">
-                {selectOptions}
-            </Select>
+            <TypeaheadSelect toggleProps={ { id: this.props.id } }
+                             onInputChange={this.debouncedChange}
+                             placeholder={placeholder}
+                             noOptionsAvailableMessage={this.state.error || _("No such file or directory")}
+                             noOptionsFoundMessage={this.state.error || _("No such file or directory")}
```

Comment:
Can these both happen? what's the difference?

Reply:
noOptionsAvailable is shown when the selectOptions property is the empty array. noOptionsFoundMessage is shown when filtering doesn't match anything.

Reply:
Ah, thanks! So would `noOptionsAvailable` happen if you select an empty directory? Then "No such  file or directory" would be confusing.